plugin.py

```diff
@@ -79,27 +144,191 @@ class TellRecord(Base):
 # session.rollback()
 session = DBSession()
 
+
 class Tell(callbacks.Plugin):
     """MemoServ replacement with extra features."""
     threaded = True
 
+    queryTell = PostTell()
+
+    pm = 0
+
+    def __init__(self, irc):
+        self.__parent = super(Tell, self)
+        self.__parent.__init__(irc)
+
+        # TODO: Load all unread messages.
+        self.queryTell.load_unread()
+
+    # Process all text before handing off to command processor
+    def inFilter(self, irc, msg):
+        # Return days, hours, minutes ago with *simple* logic
+        def get_timeago():
```

Comment:
Move this. Try and exit from the inFilter as fast as possible. This is running for every single message to the channel.

Reply:
you talking about get_timeago? It doesn't run every time, it's a nested function. inFilter needs to be there so we can get tells for anyone who types.